### PR TITLE
added new type of form field 'picker'

### PR DESCRIPTION
**Pull request by smanini1**
**Description:**
Added a new FormField type 'picker' which allows the user to create and set an array of options to a field which will be displayed in the form of a picker. 

**Preview:**

ezgif com-video-to-gif

### CODE REVIEW

File: RSFormView/Classes/Views/TextField/TextFieldViewPrivateExtension.swift [hunk 1]
```diff
@@ -74,27 +74,33 @@ internal extension TextFieldView {
...
+      datePicker.minimumDate = fieldData.minimumDate
+      datePicker.maximumDate = fieldData.maximumDate
+      datePicker.addTarget(self,
+                           action: #selector(datePickerChangedValue),
+                           for: .valueChanged)
+      textField.inputView = datePicker
+    } else {
+      let picker = UIPickerView()
+      picker.delegate = self
+      textField.inputView = picker
```
Comment (germanStabile) [inline RSFormView/Classes/Views/TextField/TextFieldViewPrivateExtension.swift]: we probably should pre-select the previously picked option if exists, same as for the date picker

File: RSFormView/Classes/Models/FormField.swift [hunk 2]
```diff
@@ -78,6 +78,9 @@ open class FormField {
   /// The maximum selectable date for *.date* FormFields
   public var maximumDate: Date?
   
+  /// Source data for *.picker* FormFields
+  public var options: [String]?
```
Comment (glm4) [inline RSFormView/Classes/Models/FormField.swift]: I think this class is getting too many responsibilities. We should refactor this to a more OO solution where we have FormField as a parent class and then PickerField, TextField, etc as subclasses. This is more like a V2 refactor, but just to raise the issue. @germanStabile WDYT?

File: Example/RSFormViewExampleApp/SignupExampleViewModel.swift [hunk 3]
```diff
@@ -143,6 +144,20 @@ class SignupExampleViewModel: FormViewModel {
...
   
+  func ageItem() -> FormItem {
+    let ageRange = 1...90
+    let stringAgeRange = ageRange.map { String($0) }
+    let ageField = FormField(name: FieldName.age.rawValue,
+                                   initialValue: "",
+                                   placeholder: FieldName.age.rawValue,
+                                   fieldType: .picker,
+                                   isValid: false,
+                                   errorMessage: "Please select the age",
```
[the remark this answers is not in the document]
Comment (smanini1) [reply Example/RSFormViewExampleApp/SignupExampleViewModel.swift]: agree!

File: RSFormView/Classes/Models/FormField.swift (hunk 2 above)
Comment (smanini1) [reply RSFormView/Classes/Models/FormField.swift]: true! we will look into creating a new card so we can take care of this!!

Comment (germanStabile) [reply RSFormView/Classes/Models/FormField.swift]: [Trello Card to tackle this](https://trello.com/c/TjB6391q/18-refactor-formfield-to-a-oo-solution-where-formfield-will-be-the-parent-class-and-different-fields-the-subclasses)

File: Example/RSFormViewExampleAppUITests/RSFormViewExampleAppUITests.swift [hunk 4]
```diff
@@ -108,6 +109,11 @@ class RSFormViewExampleAppUITests: XCTestCase {
     datePickers.pickerWheels.element(boundBy: 1).adjust(toPickerWheelValue: "13")
     datePickers.pickerWheels.element(boundBy: 2).adjust(toPickerWheelValue: "1989")
     
+    app.textFields["AGE"].tap()
+    
+    let agePicker = XCUIApplication().pickers
```
Comment (germanStabile) [inline Example/RSFormViewExampleAppUITests/RSFormViewExampleAppUITests.swift]: do app.pickers here

**Pull request merged**